src/main/java/uk/ac/ebi/atlas/hcalandingpage/HcaMetadataService.java

```diff
@@ -0,0 +1,52 @@
+package uk.ac.ebi.atlas.hcalandingpage;
+
+import com.google.common.collect.ImmutableSet;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.model.experiment.Experiment;
+import uk.ac.ebi.atlas.trader.ExperimentTrader;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+
+@Component
+public class HcaMetadataService {
+    private final HcaMetadataDao hcaMetadataDao;
+    private final ExperimentTrader experimentTrader;
+
+    public HcaMetadataService(ExperimentTrader experimentTrader,
+                              HcaMetadataDao hcaMetadataDao) {
+        this.experimentTrader = experimentTrader;
+        this.hcaMetadataDao = hcaMetadataDao;
+    }
+
+    public ImmutableSet<String> getHcaOntologyIds() {
+        var humanExperimentAccessionsAndAssociatedOntologyIds =
+                hcaMetadataDao.fetchHumanExperimentAccessionsAndAssociatedOntologyIds();
```

Comment:
No need for this throw-away variable. Chain it like so:
```java
return hcaMetadataDao
        .fetchHumanExperimentAccessionsAndAssociatedOntologyIds()
        .stream()
        ...
```

Reply:
Agreed! Made code more simple.